pkg/reconciler/revision/resources/deploy_test.go

```diff
@@ -1144,3 +1146,43 @@ func TestMakeDeployment(t *testing.T) {
 		})
 	}
 }
+
+func TestVarLogDisabled(t *testing.T) {
+	f := features.DeepCopy()
+	f.RuntimeV1VarLogVolume = config.Disabled
+
+	rev := revision("bar", "foo",
+		withContainers([]corev1.Container{{
+			Name:  servingContainerName,
+			Image: "busybox",
+			Ports: []corev1.ContainerPort{{
+				ContainerPort: 8080,
+			}},
+			ReadinessProbe: withTCPReadinessProbe(v1.DefaultUserPort),
+		}}),
+	)
+
+	// Drop all the var log additions from the serving
+	// container and pod spec
+	s := servingContainer()
+	env := s.Env[:0]
+	for _, e := range s.Env {
+		if !strings.HasPrefix(e.Name, "K_INTERNAL_") {
+			env = append(env, e)
+		}
+	}
+	s.VolumeMounts = nil
+	s.Env = env
+
+	want := podSpec([]corev1.Container{s, queueContainer()})
+	want.Volumes = nil
+
+	got, err := makePodSpec(rev, f, &logConfig, &traceConfig, &obsConfig, &deploymentConfig)
+	if err != nil {
+		t.Fatal("Got unexpected error:", err)
+	}
+
+	if diff := cmp.Diff(want, got, quantityComparer); diff != "" {
+		t.Error("MakeDeployment (-want, +got) =", diff)
```

Comment:
should this be `makePodSpec` rather than `MakeDeployment`?

Reply:
yup - thanks for the catch